### ISSUE TITLE
Automatically indent correctly when pasting or moving text

### ISSUE THREAD

**Issue by jghauser** [nvim-neorg/neorg#497]
### Issues

- [X] I have checked [existing issues](https://github.com/vhyrro/neorg/issues?q=is%3Aissue) and there are no existing ones with the same request.

### Feature description

It would be great if indent were corrected when pasting something into a section with a higher or lower level (and the same with moving text, though I personally don't use that much).

For instance:
```
* heading
** heading2
   some text
```
should become:
```
* heading
  some text
** heading2
```
rather than:
```
* heading
   some text
** heading
```

Basically, I would like for neorg to do what already happens now if i go to each pasted line and enter and exit insert mode.

### Help

No

**Comment by zwiebelbox** [comment]
Not sure if that's a good Idea. What If the indentation of the copied text is relevant to me? The vim way is to do it manually:
- In visual mode `=` to auto-indent selected lines
- In normal mode `==` to auto-indent current line
- Or after pasting: `=']` to auto-indent all pasted lines

That's the same way I would use for other programming languages:
- instead of `==` I would use `gcc` to comment the line and `gqq` to auto-format the line (when a formatexpr is set by the lsp).
- and instead of `=` in visual mode I would then use `gc` or `gq` to comment or format the selection
If you read tips like these, and you have no Idea how they work, you can always read more about keystrokes via `:help` or short `:h`. For example `:h ']`, `:h gc`, `:h =` or `:h gq`.
